spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/EndpointSubsetNS.java

```diff
@@ -16,44 +16,16 @@
 
 package org.springframework.cloud.kubernetes.fabric8.discovery;
 
-import java.util.ArrayList;
 import java.util.List;
 
 import io.fabric8.kubernetes.api.model.EndpointSubset;
 
 /**
  * @author Haytham Mohamed
  **/
-public class EndpointSubsetNS {
-
-	private String namespace;
-
-	private List<EndpointSubset> endpointSubset = new ArrayList<>();
-
-	public String getNamespace() {
-		return namespace;
-	}
-
-	public void setNamespace(String namespace) {
-		this.namespace = namespace;
-	}
-
-	public List<EndpointSubset> getEndpointSubset() {
-		return endpointSubset;
-	}
-
-	public void setEndpointSubset(List<EndpointSubset> endpointSubset) {
-		this.endpointSubset = endpointSubset;
-	}
-
-	@Override
-	public boolean equals(Object o) {
-		return this.endpointSubset.equals(o);
-	}
-
-	@Override
-	public int hashCode() {
-		return this.endpointSubset.hashCode();
+record EndpointSubsetNS(String namespace, List<EndpointSubset> endpointSubset) {
```

Comment:
I will sound like a broken record a little more, but this moves to a record.

Reply:
it took a little bit of time until I felt comfortable with : 

```
endpointSubset = endpointSubset == null ? List.of() : endpointSubset;
```

reading the bytecode was the only way for me to "get it". In essence this can be read like this:

```
List<EndpointSubset> local = endpointSubset == null ? List.of() : endpointSubset;
this.endpointSubset = local;
```

`this.endpointSubset = local` will be inserted by the compiler in a compact constructor in a record. All fields in a record are "committed" in a compact constructor at the very end and this is done by `javac`. 

---

This is not very different from a plain class constructor, for example:


```
public class Mine {

	private final List<String> endpointSubset;

	public Mine(List<String> endpointSubset) {
		if (endpointSubset == null) {
			endpointSubset = List.of();
		}
		this.endpointSubset = endpointSubset;
	}
	
}
```

But it was confusing in the beginning.